api/app/src/Migrations/Version292.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version292 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE deputy DROP CONSTRAINT FK_28FA6B9FA76ED395');
+        $this->addSql('ALTER TABLE deputy ADD CONSTRAINT FK_28FA6B9FA76ED395 FOREIGN KEY (user_id) REFERENCES dd_user (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
+        $this->addSql('ALTER TABLE document ALTER sync_attempts SET DEFAULT 0');
+        $this->addSql('ALTER TABLE pre_registration ALTER hybrid TYPE VARCHAR(12)');
+        $this->addSql('ALTER TABLE pre_registration ALTER client_firstname TYPE VARCHAR(100)');
+        $this->addSql('ALTER TABLE user_research_response DROP CONSTRAINT FK_3B9FE71A5423F28F');
+        $this->addSql('ALTER TABLE user_research_response DROP CONSTRAINT FK_3B9FE71AA76ED395');
+        $this->addSql('ALTER TABLE user_research_response ADD CONSTRAINT FK_3B9FE71A5423F28F FOREIGN KEY (researchType_id) REFERENCES research_type (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
```

Comment:
We probably want to remove this cascade delete, otherwise deleting a research type deletes the related user_research_response records.

Reply:
Looking at the research type table they are intrinsically linked to the UR submission.  This table contains the UID for the submission along with how the Survey was carried out, so if we're removing a row on this table we would want to remove the corresponding row as not to orphan it.

Reply:
Right, but my understanding is that removing a row in the research_type table would remove all the user_research_response rows which reference it. Is that the right way round? If we remove the cascade, it would prevent removing a research_type row if it was referenced from a user_research_response, I think.